backend/interf.ml

```diff
@@ -17,33 +17,35 @@
    Annotate pseudoregs with interference lists and preference lists. *)
 
 module IntPairSet =
-  Set.Make(struct
+  Hashtbl.MakeSeeded(struct
     type t = int * int
-    let compare ((a1,b1) : t) (a2,b2) =
-      match compare a1 a2 with
-        | 0 -> compare b1 b2
-        | c -> c
+
+    let equal ((a1, b1) : int * int) (a2, b2) =
+      a1 = a2 && b1 = b2
+
+    let hash = Hashtbl.seeded_hash
```

Comment:
It might be beneficial to redefine `hash`.

Reply:
What would be a good one for a pair of ints whose values are bounded by Reg.num_registers.

Reply:
My comment was actually twofold: maybe simply redefining
a function with the semantics of `Hashtbl.seeded_hash`
would be useful (like for `equal`), and indeed maybe there
is a better hash function out there for our use case.
My knowledge is too limited to be sure.

Reply:
My experiennce with https://github.com/ocaml-flambda/flambda-backend/pull/678 is that it is beneficial
to redefine the function, at least to avoid a C call.